### DISCUSSION TITLE
Training tokens of Stage1 and Stage 2

### DISCUSSION THREAD

**Discussion by kjhoon7686** [ibm-granite/discussions#16] [Q&A] [answered]
Could you let me know the number of training tokens for Stage 1 and Stage 2 respectively?

**Comment by mayank31398** [comment, marked as answer]
phase 1 is 10T, phase 2 is 2T for dense models
phase 1 is 8T, phase 2 is 2T for MoE models
you can find the details in our paper: https://github.com/ibm-granite/granite-3.0-language-models/blob/main/paper.pdf

**Reply by kjhoon7686** [reply]
Thank you for your answer. I missed that detail earlier. I was deeply impressed by your paper!
